docs/src/proposals/tiered-accounts-db-storage.md

```diff
@@ -0,0 +1,466 @@
+---
```

Comment:
Some discussion around the relative cost of accessing a hot vs cold account in a transaction would be useful. 

If there's large delta then I could imagine somebody wanting to modify their validator to optimize for only accepting transactions with hot accounts to make it more likely that the next leader will build on their block. Or protocols modifying themselves to write unnecessarily into accounts to prevent them from getting cold.

Reply:
> If there's large delta then I could imagine somebody wanting to modify their validator to optimize for only accepting transactions with hot accounts to make it more likely that the next leader will build on their block. Or protocols modifying themselves to write unnecessarily into accounts to prevent them from getting cold.

That's a good insight.  The plan is to have the size of hot and cold storage configurable so that validators with different specs/goals can be configured differently to optimize their needs.

In the end, it is a trade-off between the storage size and the run-time performance.

For preventing all validators from avoiding answering RPCs querying cold accounts, do we want to have some type of reward for validators answering cold RPC queries?